### PR TITLE
`Integrated code lifecycle`: Fix handling of empty error text nodes in XML result parser

### PR DESCRIPTION
**Pull request by just-max**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Currently, Artemis fails if test reports contain empty error/failure nodes:

```xml
<testsuites>
  <testsuite package="mwe-package" id="0" name="mwe-suite-name" timestamp="2024-08-09T12:34:56"
    hostname="localhost" tests="1" failures="1" errors="0" time="0">
    <properties></properties>
    <testcase name="mwe-name" classname="mwe-class" time="0">
      <failure type="empty"></failure>
    </testcase>
  </testsuite>
</testsuites>
```

> Error while parsing report filegrading.xml [sic], ignoring.

This is because JacksonXMLText deserializes empty text nodes as `null` instead of `""` (or even e.g. `"    "` when the node contains only whitespace). An exception is then raised by `List.of` from trying to create a `List.of` list with `null`s.

### Description

Empty error/failure nodes are instead turned into `""`.

### Steps for Testing
1. Create an exercise that writes the given XML to the test report directory
2. See that an exception is raised (before this fix)
3. After the fix, see that the result is created successfully

I have not had the chance to test this change myself.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

@coderabbitai ignore

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of XML deserialization to prevent `null` assignments in the detailed message field.
  - Ensured that empty text nodes are handled correctly, enhancing the robustness of the application.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/TestResultXmlParser.java [hunk 1]
```diff
@@ -116,7 +116,9 @@ public void setMessage(String message) {
 
         @JacksonXmlText
         public void setDetailedMessage(String detailedMessage) {
-            this.detailedMessage = detailedMessage;
+            // empty text nodes are deserialized as null instead of a string, see: https://github.com/FasterXML/jackson-dataformat-xml/issues/565
+            // note that this workaround does not fix the issue entirely, as strings of only whitespace become the empty string
+            this.detailedMessage = detailedMessage != null ? detailedMessage : "";
```
[the remark this answers is not in the document]
Comment (just-max) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/TestResultXmlParser.java:121]: No??

**Comment by just-max** [comment]
Thanks coderabbit :rofl: 

@coderabbitai ignore please

**Comment by Strohgelaender** [comment]
Changes make sense. Can you also add your example as a test?

**Review by Strohgelaender — DISMISSED** [review]

**Review by BBesrour — APPROVED** [review]
code lgtm

**Review by MichaelOwenDyer — APPROVED** [review]
Code looks good

**Review by krusche — APPROVED** [review]

**Pull request merged**